##Preparation of anotation files for the Drop-seq Pipelines



### 1. Obtain reference genomes



hg38
```bash
rsync -avzP rsync://hgdownload.cse.ucsc.edu/goldenPath/hg38/bigZips/hg38.chromFa.tar.gz .
#combine into one single drop alternative haplotypes (_alt)
tar xvzf hg38.chromFa.tar.gz
cd chroms
rm *_alt.fa
cat *.fa > ../hg38_UCSC/hg38_ucsc.fa
rm -r chroms
```



mm10

```bash
#chromFa.tar.gz - The assembly sequence in one file per chromosome.
#    Repeats from RepeatMasker and Tandem Repeats Finder (with period
#    of 12 or less) are shown in lower case; non-repeating sequence is
#    shown in upper case.

rsync -avzP rsync://hgdownload.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz .
tar xvzf chromFa.tar.gz
cat *.fa > mm10_UCSC/mm10_ucsc.fa
rm *.fa

```

#### danRer10 (zebrafish)

```bash
#Danio ri.. zebrafish

rsync -avzP rsync://hgdownload.cse.ucsc.edu/goldenPath/danRer10/bigZips/danRer10.fa.gz .

gunzip danRer10.fa.gz
```
The model system expresses additional markers that have to be included in the reference genome:

EGFP
>EGFP
ATGGTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGCGACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGCAAGCTGACCCTGAAGTTCATCTGCACCACCGGCAAGCTGCCCGTGCCCTGGCCCACCCTCGTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAGCAGCACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTCTTCAAGGACGACGGCAACTACAAGACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTGGTGAACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACGGCAACATCCTGGGGCACAAGCTGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAGAAGAACGGCATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAGCTCGCCGACCACTACCAGCAGAACACCCCCATCGGCGACGGCCCCGTGCTGCTGCCCGACAACCACTACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTGTACAAGGGCGGTGGAAGATCTGGGAATTCAAGGCCTCTCGAGCCTCTAGATTCTGCAGCCCTATAGC

mCherry
>mCherry
ATGGTGAGCAAGGGCGAGGAGGACAACATGGCCATCATCAAGGAGTTCATGCGCTTCAAGGTGCACATGGAGGGCTCCGTGAACGGCCACGAGTTCGAGATCGAGGGCGAGGGCGAGGGCCGCCCCTACGAGGGCACCCAGACCGCCAAGCTGAAGGTGACCAAGGGCGGCCCCCTGCCCTTCGCCTGGGACATCCTGTCCCCTCAGTTCATGTACGGCTCCAAGGCCTACGTGAAGCACCCCGCCGACATCCCCGACTACTTGAAGCTGTCCTTCCCCGAGGGCTTCAAGTGGGAGCGCGTGATGAACTTCGAGGACGGCGGCGTGGTGACCGTGACCCAGGACTCCTCCCTGCAGGACGGCGAGTTCATCTACAAGGTGAAGCTGCGCGGCACCAACTTCCCCTCCGACGGCCCCGTAATGCAGAAGAAGACCATGGGCTGGGAGGCCTCCTCCGAGCGGATGTACCCCGAGGACGGCGCCCTGAAGGGCGAGATCAAGCAGAGGCTGAAGCTGAAGGACGGCGGCCACTACGACGCCGAGGTCAAGACCACCTACAAGGCCAAGAAGCCCGTGCAGCTGCCCGGCGCCTACAACGTCAACATCAAGCTGGACATCACCTCCCACAACGAGGACTACACCATCGTGGAACAGTACGAGCGCGCCGAGGGCCGCCACTCCACCGGCGGCATGGACGAGCTGTACAAG


mouse c-myc
>m_cmyc
ATGCCCCTCAACGTGAACTTCACCAACAGGAACTATGACCTCGACTACGACTCCGTACAGCCCTATTTCATCTGCGACGAGGAAGAGAATTTCTATCACCAGCAACAGCAGAGCGAGCTGCAGCCGCCCGCGCCCAGTGAGGATATCTGGAAGAAATTCGAGCTGCTTCCCACCCCGCCCCTGTCCCCGAGCCGCCGCTCCGGGCTCTGCTCTCCATCCTATGTTGCGGTCGCTACGTCCTTCTCCCCAAGGGAAGACGATGACGGCGGCGGTGGCAACTTCTCCACCGCCGATCAGCTGGAGATGATGACCGAGTTACTTGGAGGAGACATGGTGAACCAGAGCTTCATCTGCGATCCTGACGACGAGACCTTCATCAAGAACATCATCATCCAGGACTGTATGTGGAGCGGTTTCTCAGCCGCTGCCAAGCTGGTCTCGGAGAAGCTGGCCTCCTACCAGGCTGCGCGCAAAGACAGCACCAGCCTGAGCCCCGCCCGCGGGCACAGCGTCTGCTCCACCTCCAGCCTGTACCTGCAGGACCTCACCGCCGCCGCGTCCGAGTGCATTGACCCCTCAGTGGTCTTTCCCTACCCGCTCAACGACAGCAGCTCGCCCAAATCCTGTACCTCGTCCGATTCCACGGCCTTCTCTCCTTCCTCGGACTCGCTGCTGTCCTCCGAGTCCTCCCCACGGGCCAGCCCTGAGCCCCTAGTGCTGCATGAGGAGACACCGCCCACCACCAGCAGCGACTCTGAAGAAGAGCAAGAAGATGAGGAAGAAATTGATGTGGTGTCTGTGGAGAAGAGGCAAACCCCTGCCAAGAGGTCGGAGTCGGGCTCATCTCCATCCCGAGGCCACAGCAAACCTCCGCACAGCCCACTGGTCCTCAAGAGGTGCCACGTCTCCACTCACCAGCACAACTACGCCGCACCCCCCTCCACAAGGAAGGACTATCCAGCTGCCAAGAGGGCCAAGTTGGACAGTGGCAGGGTCCTGAAGCAGATCAGCAACAACCGCAAGTGCTCCAGCCCCAGGTCCTCAGACACGGAGGAAAACGACAAGAGGCGGACACACAACGTCTTGGAACGTCAGAGGAGGAACGAGCTGAAGCGCAGCTTTTTTGCCCTGCGTGACCAGATCCCTGAATTGGAAAACAACGAAAAGGCCCCCAAGGTAGTGATCCTCAAAAAAGCCACCGCCTACATCCTGTCCATTCAAGCAGACGAGCACAAGCTCACCTCTGAAAAGGACTTATTGAGGAAACGACGAGAACAGTTGAAACACAAACTCGAACAGCTTCGAAACTCTGGTGCATAA



```bash
cat danRer10.fa EGFP.fa mCherry.fa cmyc.fa > danRer10_RG_cmyc.fa

```


#### panTro5

```bash

GCAGCTNTGTAGTCATCAAGGACTACGGCAAANAGTCCCAGGCTAAGGACGTCATTGAAGAG


```



### 2. Obtain transcript annotation

hg38
```bash
Release 27 (GRCh38.p10)
ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_27/gencode.v27.annotation.gtf.gz
```


mm10
```bash
Release M15 (GRCm38.p5)
ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_mouse/release_M15/gencode.vM15.annotation.gtf.gz

```

danRer10
```bash
# downloaded danRer10 from UCSC
RefGene_danRer10-2017-08-28.gtf

#in addition add the annotations for the reporter genes EGFP and mCherry, and the overexpressed oncogene c-myc
#lengths
EGFP	danRer10_refGene	CDS	1	778	.	+	.	gene_id	"EGFP";	transcript_id "EGFP";
EGFP	danRer10_refGene	exon	1	778	.	+	.	gene_id "EGFP"; transcript_id "EGFP";
mCherry	danRer10_refGene	CDS	1	708	.	+	.	gene_id "mCherry"; transcript_id "mCherry";
mCherry danRer10_refGene	exon	1	708	.	+	.	gene_id "mCherry"; transcript_id "mCherry";
m_cmyc  danRer10_refGene  CDS 1 1320  . + . gene_id "m_cmyc"; transcript_id "m_cmyc";
m_cmyc  danRer10_refGene  exon 1 1320  . + . gene_id "m_cmyc"; transcript_id "m_cmyc";

#combine into additional_sequences.gtf

cat RefGene_danRer10-2017-08-28.gtf	 additional_sequences.gtf > RefGene_danRer10-2017-08-28_RG_cmyc.gtf



```



### 3. Prepare STAR indeces

Read length is 62 bp for the current DropSeq set up. If lengths of reads changes (reads in R2 file), this needs to be changed with the appropriate length


#### hg38
```bash



```



#### Zebrafish (danRer10_RGcMyc)
This index includes the fluorochromes EGFP and mCherry as well as the oncogene c-myc (from mouse).

STAR --runThreadN 2 --runMode genomeGenerate --genomeDir dr10_noalt_juncRefGene_RG_myc_61 --genomeFastaFiles danRer10_UCSC_RG/danRer10_RG_cmyc.fa --sjdbGTFfile RefGene_danRer10-2017-08-28_RG_cmyc.gtf --sjdbOverhang 61
```
